source/basic.tex

```diff
@@ -7268,3 +7338,443 @@
+
+\pnum
+A \defn{contract violation} occurs when
+\begin{itemize}
+\item
+$B$ is \keyword{false},
+\item
+the evaluation of the predicate
+exits via an exception, or
+\item
+the evaluation of the predicate
+is performed in a context that is
+manifestly constant-evaluated\iref{expr.const}
+and the predicate
+is not a core constant expression.
+\end{itemize}
+
+\begin{note}
+If $B$ is \keyword{true},
+no contract violation occurs and
+control flow continues normally
+after the point of evaluation of the contract assertion.
+The evaluation of the predicate
+can fail to produce a value
+without causing a contract violation,
+for example,
+by calling \tcode{longjmp}\iref{csetjmp.syn}
+or terminating the program.
+\end{note}
+
+\pnum
+\indexdefn{contract evaluation semantics!terminating}%
+If a contract violation occurs
+in a context that is manifestly constant-evaluate\iref{expr.const},
+and the evaluation semantic is
+a terminating semantic,
+the program is ill-formed.
+
+\begin{note}
+A diagnostic is produced
+if the evaluation semantic is observe\iref{intro.compliance}.
+\end{note}
+
+\begin{note}
+Different evaluation semantics
+chosen for the same contract assertion
+in different translation units
+can result in
+violations of the one-definition rule\iref{basic.def.odr}
+when a contract assertion has side effects
+that alter the value produced by a constant expression.
+\begin{example}
+\begin{codeblock}
+constexpr int f(int i)
+{
+  contract_assert((++const_cast<int&>(i), true));
+  return i;
+}
+inline void g()
+{
+  int a[f(1)];  // size dependent on the evaluation semantic of \tcode{contract_assert} above
+}
+\end{codeblock}
+\end{example}
+\end{note}
+
+\pnum
+When the program is \defn{contract-terminated},
+it is
+\impldef{method by which contract termination occurs}
+(depending on context) whether
+\begin{itemize}
+\item
+\tcode{std::terminate} is called,
+\item
+\tcode{std::abort} is called, or
+\item
+execution is terminated.
+
+\begin{note}
+No further execution steps occur\iref{intro.progress}.
+\end{note}
+\end{itemize}
+
+\begin{note}
+Performing the actions of
+\tcode{std::terminate} or \tcode{std::abort}
+without actually making a library call
+is a conforming implementation of
+contract-terminating\iref{intro.abstract}.
+\end{note}
+
+\pnum
+\indextext{contract evaluation semantics!enforce}%
+\indextext{contract evaluation semantics!quick-enforce}%
+If a contract violation occurs
+in a context that is not manifestly constant-evaluated
+and the evaluation semantic is quick-enforce,
+the program is contract-terminated.
+
+\pnum
+\indextext{\idxcode{contract_violation}}%
+\indextext{contract evaluation semantics!enforce}%
+\indextext{contract evaluation semantics!observe}%
+\indexlibraryglobal{contract_violation}%
+If a contract violation occurs
+in a context that is not manifestly constant-evaluated
+and the evaluation semantic is enforce or observe,
+the contract-violation handle\iref{basic.contract.handler}
```

Comment:
Isn't it a typo?
```suggestion
the contract-violation handler\iref{basic.contract.handler}
```

Reply:
Yup, typo, thanks.  Fixed (along with some other grammar corrections pointed out offline).